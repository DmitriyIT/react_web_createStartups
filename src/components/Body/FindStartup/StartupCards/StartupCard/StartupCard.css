.Startup__popup {
	right: -105%;
	top: -50%;
	display: none;
	transition: opacity 0.1s ease;
	z-index: 20;
}
.StartupCard:hover .Startup__popup {
	display: block;
}

.StartupCard {
	position: relative;
	display: inline-block;
	width: 208px;
	box-shadow: 1px 4px 10px rgba(0, 0, 0, 0.25);
}
.StartupCard__topLine {
	height: 7px;
	background-color: #84BEBA;
}
.StartupCard__title {
	min-height: 33px;
	padding-left: 7px;
	overflow-x: hidden;

	white-space: nowrap;
	text-overflow: ellipsis;
	font-family: Fira Sans;
	font-style: italic;
	font-weight: 500;
	line-height: 33px;
	font-size: 14px;
	color: #111F24;
}
.StartupCard__body {
	padding-left: 7px;
	padding-top: 9px;
	padding-right: 30px;
	padding-bottom: 20px;
	background-color: #F3F3F3;
	font-family: Fira Sans;
	font-style: italic;
	font-weight: 500;
	line-height: normal;
	font-size: 14px;
	color: #224A55;
}
.StartupCard__bottomLine {
	height: 7px;
	background-color: #009D90;
}

.StartupCard__triangle {
	position: absolute;
	right: 0;
	bottom: 0;
}
.StartupCard__button {
	position: absolute;
	cursor: pointer;
	height: 41px;
	width: 91px;
	right: -18px;
	bottom: -10px;
	background: linear-gradient(145.33deg, #FFE485 14.26%, #009D90 431.65%);
	background-color: #F4E185;
	box-shadow: 1px 4px 10px rgba(0, 0, 0, 0.25);
	transition: background-color 1.35s;

	padding-left: 5px;
	font-family: Fira Sans;
	font-style: italic;
	font-weight: 500;
	line-height: 41px;
	font-size: 16px;
	color: #254E5B;
}
.StartupCard__button:hover {
	background: linear-gradient(145.33deg, rgba(255, 228, 133, 0.01) 14.26%, rgba(0, 157, 144, 0.01) 431.65%);
	background-color: #FFE485;
}