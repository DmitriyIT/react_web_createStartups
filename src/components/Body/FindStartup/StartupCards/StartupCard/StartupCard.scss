.Startup__popup {
	right: -105%;
	top: -50%;
	// display: none;
	visibility: hidden;
	transition: opacity 0.7s ease;
	z-index: 20;
	opacity: 0;
}
.StartupCard:hover .Startup__popup {
	// display: block;
	visibility: visible;
	opacity: 1;
}

.StartupCard {
	position: relative;
	display: inline-block;
	width: 208px;
	border-radius: 10px 10px 5px 5px;
	box-shadow: 1px 4px 10px rgba(0, 0, 0, 0.25);
}
.StartupCard__title {
	min-height: 40px;
	padding-left: 7px;
	overflow-x: hidden;
	border-radius: 10px 10px 0px 0px;
	white-space: nowrap;
	text-overflow: ellipsis;

	font-family: Fira Sans;
	font-style: italic;
	font-weight: 500;
	font-size: 14px;
	line-height: 40px;

	color: #24505F;
	background: linear-gradient(156.19deg, #FFE99C 16.29%, #FFE486 115.92%, rgba(255, 164, 91, 0.8) 166.56%);
}
.StartupCard__body {
	height: 71px;
	overflow-y: scroll;
	padding-left: 7px;
	padding-top: 9px;
	padding-right: 30px;
	padding-bottom: 20px;
	background-color: #FFF;
	font-family: Fira Sans;
	font-style: italic;
	font-weight: 500;
	line-height: normal;
	font-size: 14px;
	color: #224A55;
}
.StartupCard__bottomLine {
	height: 7px;
	border-radius: 0px 0px 5px 5px;
	background: linear-gradient(217.89deg, rgba(255, 224, 115, 0.7) 4.5%, rgba(0, 157, 144, 0.5) 91.99%);
}

.StartupCard__triangle {
	position: absolute;
	right: 0;
	bottom: 0;
}
.StartupCard__button {
	position: absolute;
	cursor: pointer;
	height: 41px;
	width: 91px;
	right: -18px;
	bottom: -10px;
	background: linear-gradient(145.33deg, #FFE485 14.26%, #009D90 431.65%);
	background-color: #F4E185;
	box-shadow: 1px 4px 10px rgba(0, 0, 0, 0.25);
	transition: background-color 1.35s;

	padding-left: 5px;
	font-family: Fira Sans;
	font-style: italic;
	font-weight: 500;
	line-height: 41px;
	font-size: 16px;
	color: #254E5B;
}
.StartupCard__button:hover {
	background: linear-gradient(145.33deg, rgba(255, 228, 133, 0.01) 14.26%, rgba(0, 157, 144, 0.01) 431.65%);
	/*background-color: #FFE485;*/
	background: linear-gradient(145.33deg, #FFE486 145.71%, #009D90 431.65%);
}

@media (max-width: 320px) {
	.StartupCard {
		display: block;
		width: 80%;
		margin-bottom: 30px;
	}
	.Startup__popup {
		right: -20%;
		top: -110%;
	}
}

@media (max-width: 400px) {
	.StartupCard {
		display: block;
		width: 70%;
		margin-bottom: 30px;
	}
	.Startup__popup {
		right: -20%;
		top: -110%;
	}
}

@media (min-width: 400px) and (max-width: 500px) {
	.StartupCard {
		display: block;
		width: 60%;
		margin-bottom: 30px;
	}
	.Startup__popup {
		right: -20%;
		top: -110%;
	}
}