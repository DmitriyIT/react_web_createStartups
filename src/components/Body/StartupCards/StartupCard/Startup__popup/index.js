import React from 'react';
import './Startup__popup.scss';

var Startup__popup = ({ author_img, author_name, need_people }) => {
	return (
		<div className="Startup__popup">
			<div className="popup__main-div">
				<div className="popup__author">
					<p>Автор: </p>
					<img src={'../../img/icon/' + author_img} />
					<p className="popup__author-name"> {author_name} </p>
				</div>
				<div className="popup__need">
					<p>Нужны:</p>
					<ul>
						{need_people.map(item => (<li> {item.position} {item.isFound ? '(найден)' : ''} </li>))}
					</ul>
				</div>
			</div>
			<div className="popup__back-div" />
		</div>
	);
};

export default Startup__popup;