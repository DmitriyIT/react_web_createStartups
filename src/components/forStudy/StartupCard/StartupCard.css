.StartupCard {
	background-color: grey;
	width: 300px;
	height: 150px;
	padding-left: 20px;
	margin-bottom: 20px;
}
.StartupCard_active {
	background-color: green;
}
.StartupCard__titel {
	
}
.StartupCard__text {
	
}