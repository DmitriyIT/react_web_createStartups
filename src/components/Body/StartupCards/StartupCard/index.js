import React from 'react';
import './StartupCard.css';

var StartupCard = ({ title, body, id }) => {
	return (
		<div className="StartupCard">
			<div className="StartupCard__topLine"/>
			<div className="StartupCard__title"> {title} </div>
			<div className="StartupCard__body"> {body} </div>
			<div className="StartupCard__bottomLine"/>
			<img className="StartupCard__triangle" src="../../img/triangle.svg" />
			<div className="StartupCard__button">Просмотр</div>
		</div>
	);
};

export default StartupCard;