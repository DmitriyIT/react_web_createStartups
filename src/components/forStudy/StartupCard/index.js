import React, {Component} from 'react';
import './StartupCard.css';

class StartupCard extends Component {
	constructor(props) {
		super(props);
		this.state = {
			click: false
		};
	}

	pressButton = (e) => {
		console.log(e);
		this.setState( function(prevState, props) {
			return {click: !prevState.click};
		});
	}

	render() {
		var d = this.state.click ? ' StartupCard_active' : '';
		return (
			<div className={"StartupCard" + d}>
				<img src="" />
				<p className="StartupCard__titel"> {this.props.titel} </p>
				<p className="StartupCard__text"> {this.props.text} </p>
				<button onClick={this.pressButton}> push me )</button>
			</div>
		);
	}
}

StartupCard.defaultProps = {
	arr: [{titel: "sorry", text: 'content is non'}]
};

export default StartupCard;