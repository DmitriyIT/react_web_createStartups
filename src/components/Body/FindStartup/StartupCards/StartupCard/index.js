import React from 'react';
import { Link } from 'react-router-dom';
import './StartupCard.scss';
import Startup__popup from './Startup__popup';

import PropTypes from 'prop-types';

var StartupCard = ({ title, body, id, more_info }) => {
	return (
		<div className="StartupCard">
			<div className="StartupCard__title"> {title} </div>
			<div className="StartupCard__body"> {body} </div>
			<div className="StartupCard__bottomLine"/>

			<img className="StartupCard__triangle" src="../../img/triangle.svg" />
			<Link to={"/showStartup/" + id}>
				<div className="StartupCard__button">Просмотр</div>
			</Link>
			<Startup__popup {...more_info} />
		</div>
	);
};

StartupCard.propTypes = {
	title: PropTypes.string.isRequired,
	body: PropTypes.string, 
	id: PropTypes.number,
	more_info: PropTypes.object
};
StartupCard.defaultProps = {
	title: "default title ')"
};

export default StartupCard;