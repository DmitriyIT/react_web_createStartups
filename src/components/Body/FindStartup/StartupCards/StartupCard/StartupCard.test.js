import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import StartupCard from './index.js';


it("make snapshot ", ()=> {
	var more_info = {
		author_img: 'people.svg', 
		author_name: 'Вася Вася',
		need_people: [
			{
				position: 'программист JS',
				isFound: true
			},
			{
				position: 'маркетолог',
				isFound: false
			}
		]
	};
	var props = {
		title: 'title smth', 
		body: 'body text', 
		id: 21, 
		more_info: more_info
	};

	var elem = renderer.create(
		<BrowserRouter>
			<StartupCard {...props} />
		</BrowserRouter>
	);
	var tree = elem.toJSON();
	expect(tree).toMatchSnapshot();
});
