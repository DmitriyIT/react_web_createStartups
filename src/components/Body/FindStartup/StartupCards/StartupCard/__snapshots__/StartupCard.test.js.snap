// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`make snapshot  1`] = `
<div
  className="StartupCard"
>
  <div
    className="StartupCard__title"
  >
     
    title smth
     
  </div>
  <div
    className="StartupCard__body"
  >
     
    body text
     
  </div>
  <div
    className="StartupCard__bottomLine"
  />
  <img
    className="StartupCard__triangle"
    src="../../img/triangle.svg"
  />
  <a
    href="/showStartup/21"
    onClick={[Function]}
  >
    <div
      className="StartupCard__button"
    >
      Просмотр
    </div>
  </a>
  <div
    className="Startup__popup"
  >
    <div
      className="popup__main-div"
    >
      <div
        className="popup__author"
      >
        <p>
          Автор: 
        </p>
        <img
          src="../../img/icon/people.svg"
        />
        <p
          className="popup__author-name"
        >
           
          Вася Вася
           
        </p>
      </div>
      <div
        className="popup__need"
      >
        <p>
          Нужны:
        </p>
        <ul>
          <li>
             
            программист JS
             
            (найден)
             
          </li>
          <li>
             
            маркетолог
             
            
             
          </li>
        </ul>
      </div>
    </div>
    <div
      className="popup__back-div"
    />
  </div>
</div>
`;
